clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -83,24 +85,35 @@ func (*waitForControlledPodsRunningMeasurement) Execute(config *measurement.Meas
 	if err != nil {
 		return summaries, err
 	}
+	var wg wait.Group
+	errList := util.NewErrorList()
+	var lock sync.Mutex
 	runningRuntimeObjects := 0
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(defaultWaitForRCPodsInterval) {
-		if runningRuntimeObjects >= len(runtimeObjectsList) {
-			return summaries, nil
-		}
-
-		if err := waitForRuntimeObject(config.ClientSet, runtimeObjectsList[runningRuntimeObjects], timeout-time.Since(start)); err != nil {
-			return summaries, fmt.Errorf("waiting for %v error: %v", kind, err)
-		}
-		runningRuntimeObjects++
-		if namespace == metav1.NamespaceAll {
-			glog.Infof("%s: %d / %d %ss with all pods running", waitForControlledPodsRunningName, runningRuntimeObjects, len(runtimeObjectsList), kind)
-		} else {
-			glog.Infof("%s: %v: %d / %d %ss with all pods running", waitForControlledPodsRunningName, namespace, runningRuntimeObjects, len(runtimeObjectsList), kind)
-		}
-	}
-
-	return summaries, fmt.Errorf("timeout while waiting for %ss to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found running with all pods", kind, namespace, labelSelector, fieldSelector, runningRuntimeObjects)
+	for objectCouter := 0; objectCouter < len(runtimeObjectsList); objectCouter++ {
+		objectIndex := objectCouter
+		wg.Start(func() {
+			if err := waitForRuntimeObject(config.ClientSet, runtimeObjectsList[objectIndex], timeout); err != nil {
+				errList.Append(fmt.Errorf("waiting for %v error: %v", kind, err))
+				return
+			}
+
+			lock.Lock()
```

Comment:
Please use [atomic counters](https://gobyexample.com/atomic-counters) instead of manually managing all this locking.

Reply:
This lock is not only for runningRuntimeObjects variable, but also for logging.
I agree that if I remove the lock and use atomic counter, the runningRuntimeObjects will be calculated correctly. However, if I remove lock from logging part, it is possible that variable will change in a meantime. Potential can log same number of running objects multiple times.
WYT?

Reply:
You're right, but I don't think that's a big problem. You can for e.g fix the log by just printing the current RC instead of the count till then.

Reply:
According to what we discussed offline, replaced lock with atomic counter.
Only name of a ready object is printed to log.